static/js/index.js

```diff
@@ -389,6 +394,21 @@
     newSearch(true, true);
   };
 
+  eltLogo.onclick = function() {
+    // Restore the state for "full" layout
+    setPageLayout("full");
+    eltSearchInput.value = "";
+
+    // Append the language info to the current url
+    window.history.pushState("", "", "?g=" + eltLang.value);
```

Comment:
Should add `if (window.history && window.history.pushState)`. If the browser does not support, can we `window.location.href="/"`?